python/lsst/obs/base/_fitsRawFormatterBase.py

```diff
@@ -384,7 +402,27 @@ def readFull(self, parameters=None):
         full.setMetadata(self.metadata)
         return full
 
-    def readRawHeaderWcs(self, parameters=None):
+    def read(self, component=None):
+        # Docstring inherited.
+        parameters = self.fileDescriptor.parameters
+        if parameters is None:
+            parameters = {}
+        if "bbox" in parameters:
+            raise TypeError("Raw formatters do not support reading arbitrary subimages, as some "
+                            "implementations may be assembled on-the-fly.")
+        if self.fileDescriptor.readStorageClass != self.fileDescriptor.storageClass:
```

Comment:
This check is something I'd like to see move into the base `Formatter` or `FileDatastore` on DM-29964, letting concrete `Formatters` just dispatch on what `component` is.